packages/did-document/src/lite/index.ts

```diff
@@ -1,68 +1,34 @@
-import { IDIDDocument, IResolver } from '@ew-did-registry/did-resolver-interface';
+import {
+  IDIDDocument, IResolver, DelegateTypes, IPublicKey, IServiceEndpoint, IAuthentication
+} from '@ew-did-registry/did-resolver-interface';
 import { IDIDDocumentLite } from './interface';
 
 class DIDDocumentLite implements IDIDDocumentLite {
   /**
-   * Resolver that fetches DID Document
-   */
-  private readonly resolver: IResolver;
+* @param {string} did
+* @param {Resolver} resolver
+*/
+  // eslint-disable-next-line no-useless-constructor
+  constructor(public did: string, private resolver: IResolver) { }
 
-  /**
-   * DID of concern
-   */
-  did: string;
-
-  /**
-   * Fetched DID Document
-   */
-  didDocument: IDIDDocument;
+  identityOwner(did = this.did): Promise<string> {
+    return this.resolver.identityOwner(did);
+  }
 
-  /**
-   * Constructor takes DID of interest and Resolver as inputs
-   * @param {string} did
-   * @param {Resolver} resolver
-   */
-  constructor(did: string, resolver: IResolver) {
-    this.resolver = resolver;
-    this.did = did;
+  validDelegate(
```

Comment:
handle the `Promise` with `async`

Reply:
in this case, async is not necessary since await is not used in the function

Reply:
see comment above